### ISSUE TITLE
New scipy test fatal error after switch to meson

### ISSUE THREAD

**Issue by lesteve** [pyodide/pyodide#4310]
## 🐛 Bug

There is a new fatal error in scipy tests after switch to meson i.e. #4276. I double-checked that the error does not happen before #4276. Not sure why I did not spot this when I tried to run the tests against the PR.

### To Reproduce

```py
import numpy as np
from numpy import eye, random
from scipy.linalg import logm

random.seed(1234)
dtype = np.float64
n = 2
scale = 1e-4
A = (eye(n) + random.rand(n, n) * scale).astype(dtype)
logm(A)
```

### Expected behavior

No error

### Additional context

Here is the stack-trace:
```
RuntimeError: null function or function signature mismatch
    at _dop.cpython-311-wasm32-emscripten.so.ndarray_from_pyobj (VM230 _dop.cpython-311-wasm32-emscripten.so-0004011e:1:44648)
    at invoke_iiiiiiii (VM170 pyodide.asm.js:34381:36)
    at _flapack.cpython-311-wasm32-emscripten.so.f2py_rout__flapack_dgees (VM250 _flapack.cpython-311-wasm32-emscripten.so-00601166:1:242609)
    at _dop.cpython-311-wasm32-emscripten.so.fortran_call (VM230 _dop.cpython-311-wasm32-emscripten.so-0004011e:1:48030)
    at _PyEMJS_TrampolineCall (VM170 pyodide.asm.js:1722:86)
    at pyodide.asm.wasm._PyEM_TrampolineCall (VM174 pyodide.asm.wasm-025ba86e:1:2992131)
    at pyodide.asm.wasm._PyObject_MakeTpCall (VM174 pyodide.asm.wasm-025ba86e:1:1717027)
    at pyodide.asm.wasm.PyObject_Vectorcall (VM174 pyodide.asm.wasm-025ba86e:1:1718548)
    at pyodide.asm.wasm._PyEval_EvalFrameDefault (VM174 pyodide.asm.wasm-025ba86e:1:2532710)
    at pyodide.asm.wasm._PyEval_Vector (VM174 pyodide.asm.wasm-025ba86e:1:2549198)
main @ VM165 scipy-pytest.js:58
```

Debugging a bit it does seem that inside the `ndarray_from_pyobj` wasm code there is a `call indirect` with a null function.

In other words:
```
pyodide._module.wasmTable.get(stack[1].value)
null
```

**Comment by ryanking13** [comment]
Thanks for the report!

I compared the `ndarray_from_pyobj` from Pyodide 0.24.1 and main branch, and one difference I found is:

In Pyodide 0.24.1:

```
local.get $var0
global.get $GOT.mem._npy_f2py_ARRAY_API
i32.load
i32.load offset=180
call_indirect (param i32) (result i32)
```

In the main branch:

```
local.get $var0
global.get $global36
i32.load
i32.load offset=180
call_indirect (param i32) (result i32)
```

So it seems like the symbol `_npy_f2py_ARRAY_API` is not available after the PR #4276. I am not sure if it is the main reason for the failure though.

**Comment by lesteve** [comment]
Thanks for your answer, let me know if I can help!

**Comment by ryanking13** [comment]
Another interesting behavior is that if I import `scipy.integrate` before running the `logm(A)`, then the error doesn't happen.

```python
import numpy as np
from numpy import eye, random
from scipy.linalg import logm
import scipy.integrate  # Added!

random.seed(1234)
dtype = np.float64
n = 2
scale = 1e-4
A = (eye(n) + random.rand(n, n) * scale).astype(dtype)
logm(A)
```

So it seems like some symbols that are loaded when we import `scipy.integrate` are not available... but it is very strange that it must be imported in some place anyway.

**Comment by lesteve** [comment]
Interesting indeed :thinking: ...

Just curious how did you figure out `scipy.integrate` was setting the symbol?

**Comment by ryanking13** [comment]
> Just curious how did you figure out scipy.integrate was setting the symbol?

I added the failing test to `test_scipy.py` and noticed that it passed, which means that running other tests before the failing test makes the failing test success.

So I did some random experiments after that.
